src/plugins/intel_gpu/src/graph/graph_optimizer/post_optimize_weights.cpp

```diff
@@ -75,7 +75,8 @@ void post_optimize_weights::optimize_weights(T& node, program& p) {
             bool can_be_fused = prev_node.is_type<reorder>() &&
                                 prev_node.as<reorder>().is_simple_reorder() &&
                                 prev_node.get_users().size() == 1 &&
-                                prev_node.get_dependencies().size() == 1;
+                                prev_node.get_dependencies().size() == 1 &&
+                                is_supported_weights_layout(prev_node.get_input_layout().format);
```

Comment:
we already have a function to check weight format. https://github.com/openvinotoolkit/openvino/blob/e4c7194cb31461e06d8d12c81dc9466db4032b7b/src/plugins/intel_gpu/include/intel_gpu/runtime/format.hpp#L275

Reply:
What you suggested looks better.